lib/wit/package.py

```diff
@@ -55,16 +58,17 @@ def add_dependent(self, dep):
         if dep not in self.dependents:
             self.dependents.append(dep)
 
-    def load_repo(self, wsroot, download=False, needed_commit=None):
+    def load(self, wsroot, download, source=None, revision=None):
         """Connect a Package to a GitRepo on disk.
 
         If found, self.repo will be updated.
         """
 
-        needed_commit = needed_commit or self.revision
+        source = self.resolve_source(source) or self.resolve_source(self.source)
+        revision = revision or self.revision
 
-        if needed_commit is None:
-            raise Exception("Cannot load repo for unknown commit.")
+        assert revision is not None, WitBug("Cannot load repo for unknown commit.")
+        assert source is not None, WitBug("Cannot load repo for unknown source.")
```

Comment:
Does this throw a `WitBug`? I wasn't aware of this syntax, nice!

Reply:
I just tried this out locally with a toy example, and no, it does not. It still throws an `AssertionError`. The [Python docs](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement) say that this is equivalent to `raise AssertionError(WitBug("foo"))`.

Reply:
Correct. However, WitBug being the type of the error message for AssertionError helps the user know that if they see that stacktrace, it's something they should report

Reply:
Better idea: I'll make the assertion error message a string, catch AssertionError in `main.py`, then wrap it with `WitBug` before reraising